lib/rubygems/util.rb

```diff
@@ -112,4 +112,50 @@ def self.correct_for_windows_path(path)
     end
   end
 
+  class DefaultKey
+  end
+
+  class PrivateType
+  end
+
+  require "date"
```

Comment:
This seems a problem (RubyGems shouldn't load as few stdlibs as possible), but it could probably be worked around easily (e.g., check `obj.class.name` instead)